src/commonMain/kotlin/fr/acinq/lightning/channel/Commitments.kt

```diff
@@ -705,8 +705,10 @@ data class Commitments(
     }
 
     fun receiveCommit(commits: List<CommitSig>, keyManager: KeyManager, log: MDCLogger): Either<ChannelException, Pair<Commitments, RevokeAndAck>> {
-        // first we make sure that we have exactly one commit_sig for each active commitment
-        if (commits.size != active.size) {
+        // We may receive more commit_sig than the number of active commitments, because there can be a race where we send splice_locked
+        // while our peer is sending us a batch of commit_sig. When that happens, we simply need to discard the commit_sig that belong
+        // to commitments we deactivated.
+        if (commits.size < active.size) {
             return Either.Left(CommitSigCountMismatch(channelId, active.size, commits.size))
         }
         val active1 = active.zip(commits).map {
```

Comment:
There are two subtelties that make it work: ordering of sigs and behavior of `zip`.

```suggestion
        // Signatures are sent in order (most recent first), calling `zip` will drop trailing sigs that are for deactivated/pruned commitments.
        val active1 = active.zip(commits).map {
```

Reply:
I thought the behavior of zip was standard enough, but an explicit comment is always good :+1: